ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -385,16 +384,14 @@ private boolean foundCurrentOrFailedCompactions(ShowCompactResponse compactions,
       LOG.warn("Will not initiate compaction for " + ci.getFullPartitionName() + " since last " +
         MetastoreConf.ConfVars.COMPACTOR_INITIATOR_FAILED_THRESHOLD + " attempts to compact it failed.");
 
-      ci.errorMessage = "Compaction is not initiated since last " +
-        MetastoreConf.ConfVars.COMPACTOR_INITIATOR_FAILED_THRESHOLD + " consecutive compaction attempts failed)";
-
-      txnHandler.markFailed(ci);
+      txnHandler.markFailed(ci.toFailedCompacion("Compaction is not initiated since last " +
+          MetastoreConf.ConfVars.COMPACTOR_INITIATOR_FAILED_THRESHOLD + " consecutive compaction attempts failed)"));
       return true;
     }
     return false;
   }
 
-  private CompactionType checkForCompaction(final CompactionInfo ci,
+  private CompactionRequest checkForCompaction(final CompactionCandidate ci,
```

Comment:
that is confusing method naming. checkForCompaction answers what type of compaction is needed if any, but doesn't generate the compaction request. request should be created only if compaction is required
````

Reply:
Removed all CompactionInfo class structure related changes